torch/utils/cpp_extension.py

```diff
@@ -524,8 +524,8 @@ def build_extensions(self) -> None:
             if 'nvcc_dlink' in extension.extra_compile_args:
                 assert self.use_ninja, f"With dlink=True, ninja is required to build cuda extension {extension.name}."
 
-        # Register .cu, .cuh and .hip as valid source extensions.
-        self.compiler.src_extensions += ['.cu', '.cuh', '.hip']
+        # Register .cu, .cuh, .hip, and .mm as valid source extensions.
+        self.compiler.src_extensions += ['.cu', '.cuh', '.hip', '.mm']
```

Comment:
Why not guard those by `if sys.platform == 'darwin'` or `if torch.backends.mps.is_built`?

Reply:
From my understanding the purpose of this list is to allow the extensions to be compiled only. Whether they will actually be compiled is contingent upon whether they're specified as the source files in a Python extension or not. But let's see if the CI complains.

Reply:
Added `if torch.backends.mps.is_built` guard as a precaution against something possibly failing in the internal builds at Meta.